File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/RouteStatusLabelsTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import kotlin.test.Test
import kotlin.test.assertEquals

class RouteStatusLabelsTest {
  @Test
  fun `maps known route statuses to readable labels`() {
    assertEquals("Aberta", routeStatusLabel("ABERTA"))
    assertEquals("Em andamento", routeStatusLabel("EM_ANDAMENTO"))
    assertEquals("Finalizada", routeStatusLabel("FINALIZADA"))
    assertEquals("Cancelada", routeStatusLabel("CANCELADA"))
  }

  @Test
  fun `formats unknown route statuses without raw underscores`() {
    assertEquals("Pausada temporaria", routeStatusLabel("PAUSADA_TEMPORARIA"))
  }

  @Test
  fun `uses a safe fallback for blank route statuses`() {
    assertEquals("Status desconhecido", routeStatusLabel("   "))
  }
}

File: apps/motoboy-pwa/src/routeStatusLabels.test.ts
import { describe, it } from "node:test";
import assert from "node:assert/strict";
import { routeStatusLabel } from "./routeStatusLabels";

describe("routeStatusLabel", () => {
  it("maps known route statuses to readable labels", () => {
    assert.equal(routeStatusLabel("ABERTA"), "Aberta");
    assert.equal(routeStatusLabel("EM_ANDAMENTO"), "Em andamento");
    assert.equal(routeStatusLabel("FINALIZADA"), "Finalizada");
    assert.equal(routeStatusLabel("CANCELADA"), "Cancelada");
  });

  it("formats unknown route statuses without raw underscores", () => {
    assert.equal(routeStatusLabel("PAUSADA_TEMPORARIA"), "Pausada temporaria");
  });

  it("uses a safe fallback for blank route statuses", () => {
    assert.equal(routeStatusLabel("   "), "Status desconhecido");
  });
});
